pkg/apis/storage/types.go

```diff
@@ -672,3 +672,48 @@ type CSIStorageCapacityList struct {
 	// Items is the list of CSIStorageCapacity objects.
 	Items []CSIStorageCapacity
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeAttributesClass represents a class of volume attributes. It holds the mutable
+// attributes of volumes for the provisioner that should create and update volumes.
+//
+// VolumeAttributesClasses are non-namespaced; the name of the volume attributes class
+// according to etcd is in ObjectMeta.Name.
+type VolumeAttributesClass struct {
+	metav1.TypeMeta
+
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta
+
+	// Name of the csi driver
+	DriverName string
+
+	// Parameters holds mutable volume attributes for the csi dirver. i.e. iops,
```

Comment:
parameters hold volume attributes defined by the CSI driver.  These values are opaque to the system and are passed directly to the CSI driver. The underlying storage provider supports changing these attributes on an existing volume, however the parameters field itself is immutable. To invoke a volume update, a new VolumeAttributesClass should be created with new parameters, and the PersistentVolumeClaim should be updated to reference the new VolumeAttributesClass.

The keys cannot be empty, and the maximum number of parameters is 512, with a cumulative max size of 256K.